Crash on peer list all --details

I've been trying to get the data of the `track.ExtendedPeerstore` by calling `GetAllData()` and Rumor is crashing with a panic error. I've been able to replicate the same panice error by calling the command `peer list all --details`.

Error logs when calling the `GetAllData()` from my code:
```
panic: interface conversion: interface {} is beacon.Status, not *beacon.Status

goroutine 354 [running]:
github.com/protolambda/rumor/p2p/track/dstrack.(*dsStatusBook).Status(0xc000260180, 0xc0096d1c80, 0x27, 0xc00a164150)
        /home/cortze/cortze/rumor/p2p/track/dstrack/status.go:59 +0xf6
github.com/protolambda/rumor/p2p/track/dstrack.(*dsExtendedPeerstore).GetAllData(0xc00667a4b0, 0xc0096d1c80, 0x27, 0x34)
        /home/cortze/cortze/rumor/p2p/track/dstrack/dstrack.go:229 +0x581
github.com/protolambda/rumor/metrics.(*GossipState).MarshalPeerStore(0xc0001640e0, 0x7fe6e7584eb0, 0xc000112720, 0x900, 0x0, 0x0, 0x4a3bec, 0x1fa3560)
        /home/cortze/cortze/rumor/metrics/gossip-metrics.go:81 +0x14f
github.com/protolambda/rumor/metrics.(*GossipState).ExportMetrics(0xc0001640e0, 0xc0066f266c, 0x1b, 0x0, 0x0, 0x7fe6e7584eb0, 0xc000112720, 0x0, 0xc0082f06c0)
        /home/cortze/cortze/rumor/metrics/gossip-metrics.go:93 +0x95
github.com/protolambda/rumor/control/actor/gossip.(*GossipExportMetricsCmd).Run.func1(0xc0066e7410, 0xc006f2c0a0)
        /home/cortze/cortze/rumor/control/actor/gossip/export_metris.go:44 +0x111
created by github.com/protolambda/rumor/control/actor/gossip.(*GossipExportMetricsCmd).Run
        /home/cortze/cortze/rumor/control/actor/gossip/export_metris.go:35 +0x75
``` 
Error logs when calling the `peer list all --details` from my code:
```
» mike: peer list all --details
panic: interface conversion: interface {} is beacon.Status, not *beacon.Status

goroutine 1 [running]:
github.com/protolambda/rumor/p2p/track/dstrack.(*dsStatusBook).Status(0xc000966180, 0xc00ad2cba0, 0x27, 0xc008deaca0)
        /home/cortze/protolambda/rumor/p2p/track/dstrack/status.go:59 +0xf6
github.com/protolambda/rumor/p2p/track/dstrack.(*dsExtendedPeerstore).GetAllData(0xc00671fae0, 0xc00ad2cba0, 0x27, 0x34)
        /home/cortze/protolambda/rumor/p2p/track/dstrack/dstrack.go:229 +0x581
github.com/protolambda/rumor/control/actor/peer.(*PeerListCmd).Run(0xc0095f51a0, 0x154e4c0, 0xc0095f4f00, 0xc00ab579d0, 0x0, 0x1, 0x0, 0x1)
        /home/cortze/protolambda/rumor/control/actor/peer/list.go:44 +0x1c1
github.com/protolambda/ask.(*CommandDescription).Execute(0xc0071d8480, 0x154e4c0, 0xc0095f4f00, 0xc006c78850, 0x2, 0x2, 0x0, 0x28, 0xc0095f5170, 0xc000100380)
        /home/cortze/go/pkg/mod/github.com/protolambda/ask@v0.0.5/ask.go:323 +0x606
github.com/protolambda/ask.(*CommandDescription).Execute(0xc0071d8400, 0x154e4c0, 0xc0095f4f00, 0xc006c78840, 0x3, 0x3, 0x0, 0x10bc060, 0xc000061428, 0xc0082f04c8)
        /home/cortze/go/pkg/mod/github.com/protolambda/ask@v0.0.5/ask.go:270 +0xb23
github.com/protolambda/ask.(*CommandDescription).Execute(0xc0071d8380, 0x154e4c0, 0xc0095f4f00, 0xc006c78830, 0x4, 0x4, 0x1, 0x4, 0x0, 0x0)
        /home/cortze/go/pkg/mod/github.com/protolambda/ask@v0.0.5/ask.go:270 +0xb23
github.com/protolambda/rumor/control.(*SessionProcessor).MakeCall(0xc0003a8000, 0x154e4c0, 0xc0095f4f00, 0x1530660, 0xc0003960e0, 0xc00a1140c0, 0x4, 0xc00a114148, 0x4, 0xc006c78830, ...)
        /home/cortze/protolambda/rumor/control/session_processor.go:243 +0x819
github.com/protolambda/rumor/control.(*Session).runCmd(0xc0003b4000, 0x154e4c0, 0xc0095f4f00, 0xc006c78820, 0x5, 0x5, 0x0, 0x0)
        /home/cortze/protolambda/rumor/control/session.go:383 +0x1d43
github.com/protolambda/rumor/control.(*Session).RunCmd(0xc0003b4000, 0x154e4c0, 0xc0095f4f00, 0xc006c78820, 0x5, 0x5, 0x0, 0x0)
        /home/cortze/protolambda/rumor/control/session.go:170 +0x84
mvdan.cc/sh/v3/interp.(*Runner).exec(0xc0003b6000, 0x154e440, 0xc00003a198, 0xc006c78820, 0x5, 0x5)
        /home/cortze/go/pkg/mod/mvdan.cc/sh/v3@v3.1.2/interp/runner.go:736 +0x88
mvdan.cc/sh/v3/interp.(*Runner).call(0xc0003b6000, 0x154e440, 0xc00003a198, 0x1000300000034, 0xc006c78820, 0x5, 0x5)
        /home/cortze/go/pkg/mod/mvdan.cc/sh/v3@v3.1.2/interp/runner.go:732 +0x404
mvdan.cc/sh/v3/interp.(*Runner).cmd(0xc0003b6000, 0x154e440, 0xc00003a198, 0x1549b80, 0xc0004240a0)
        /home/cortze/go/pkg/mod/mvdan.cc/sh/v3@v3.1.2/interp/runner.go:335 +0x1696
mvdan.cc/sh/v3/interp.(*Runner).stmtSync(0xc0003b6000, 0x154e440, 0xc00003a198, 0xc0004180b0)
        /home/cortze/go/pkg/mod/mvdan.cc/sh/v3@v3.1.2/interp/runner.go:275 +0x312
mvdan.cc/sh/v3/interp.(*Runner).stmt(0xc0003b6000, 0x154e440, 0xc00003a198, 0xc0004180b0)
        /home/cortze/go/pkg/mod/mvdan.cc/sh/v3@v3.1.2/interp/runner.go:256 +0x1e9
mvdan.cc/sh/v3/interp.(*Runner).Run(0xc0003b6000, 0x154e440, 0xc00003a198, 0x153ef40, 0xc0004180b0, 0xc00ac3cbc0, 0xc000396040)
        /home/cortze/go/pkg/mod/mvdan.cc/sh/v3@v3.1.2/interp/api.go:525 +0x261
github.com/protolambda/rumor/control.(*Session).Run(0xc0003b4000, 0x154e440, 0xc00003a198, 0x153ef40, 0xc0004180b0, 0xe37b7c, 0x0)
        /home/cortze/protolambda/rumor/control/session.go:108 +0x5b
github.com/protolambda/rumor/sh.ShellCmd.func2.1.1(0xc000426000, 0x1, 0x1, 0x1)
        /home/cortze/protolambda/rumor/sh/shell.go:52 +0x10b
mvdan.cc/sh/v3/syntax.(*Parser).Interactive.func1(0xc0004180b0, 0x1)
        /home/cortze/go/pkg/mod/mvdan.cc/sh/v3@v3.1.2/syntax/parser.go:187 +0xa0
mvdan.cc/sh/v3/syntax.(*Parser).stmts(0xc0003b2a80, 0xc0082f1b58, 0x0, 0x0, 0x0)
        /home/cortze/go/pkg/mod/mvdan.cc/sh/v3@v3.1.2/syntax/parser.go:799 +0xb0
mvdan.cc/sh/v3/syntax.(*Parser).Stmts(0xc0003b2a80, 0x152f720, 0xc0003a00c0, 0xc0002efb58, 0xc0003b2a80, 0xc0002efb78)
        /home/cortze/go/pkg/mod/mvdan.cc/sh/v3@v3.1.2/syntax/parser.go:114 +0xba
mvdan.cc/sh/v3/syntax.(*Parser).Interactive(0xc0003b2a80, 0x152eb60, 0xc000396140, 0xc00039a1e0, 0x0, 0x126a08e)
        /home/cortze/go/pkg/mod/mvdan.cc/sh/v3@v3.1.2/syntax/parser.go:181 +0xd4
github.com/protolambda/rumor/sh.ShellCmd.func2.1(0xc000396080, 0xc000396060)
        /home/cortze/protolambda/rumor/sh/shell.go:45 +0x150
github.com/protolambda/rumor/sh.shellMode(0xc000396000, 0xc0002efce0)
        /home/cortze/protolambda/rumor/sh/util.go:135 +0xf8
github.com/protolambda/rumor/sh.ShellCmd.func2(0xc000315340, 0x20c1708, 0x0, 0x0)
        /home/cortze/protolambda/rumor/sh/shell.go:39 +0x259
github.com/spf13/cobra.(*Command).execute(0xc000315340, 0x20c1708, 0x0, 0x0, 0xc000315340, 0x20c1708)
        /home/cortze/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0xc000314580, 0xc0002eff48, 0x6, 0x6)
        /home/cortze/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
        /home/cortze/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
        /home/cortze/protolambda/rumor/main.go:18 +0x151
```

--- a/p2p/track/dstrack/status.go
+++ b/p2p/track/dstrack/status.go
@@ -36,7 +36,7 @@ func (sb *dsStatusBook) loadStatus(p peer.ID) (*beacon.Status, error) {
 		return nil, fmt.Errorf("failed parse status bytes from datastore: %v", err)
 	}
 	// cache it
-	sb.data.Store(p, status)
+	sb.data.Store(p, &status)
 	return &status, nil
 }
 

